### DISCUSSION TITLE
Waiting for modem in boothold mode...

### DISCUSSION THREAD

**Discussion by diewet** [danielewood/sierra-wireless-modems#69] [Q&A]
I have a Dell EM7455 modem and try to flash the generic firmware version using your autoflash script. I run the script on Ubuntu 18 LTS on a Jetson Xavier.

The script recognizes the modem:

```
Searching for EM7455/MC7455 USB modems...
---
Found EM7455/MC7455: 
    Bus 002 Device 003: ID 413c:81b6 Dell Computer Corp.
```

But when the flashing process starts, I get the following error:

```
Flashing SWI9X30C_02.33.03.00.cwe onto Generic Sierra Modem...
error: reseter operation finished: No devices found to run reset operation
Waiting for modem in boothold mode...
```

Does anybody know what is going wrong?

**Comment by diewet** [comment]
I discovered an additional error when the script runs mode switch:

```
Running Modem Mode Switch to usbcomp=8
Option device requires an argument
Usage: ./swi_setusbcomp.pl [options]  

Where [options] are
  --device=<dev>        use <dev> for MBIM or QMI commands (default: '/dev/cdc-wdm0')
  --usbcomp=<num>	change USB composition setting
  --reset		issue a QMI reset request
  --usbreset		USB device reset - might be necessary for MC74xx
  --qdl                 reboot modem into bootloader QDL mode
  --debug		enable verbose debug output
  --help		this help text

  The current setting and supported modes will always be displayed
```

Looks like there is a problem in the script with the line:

`./swi_setusbcomp.pl --usbcomp=$swi_usbcomp --device="$devpath"`

**Comment by GerritErpenstein** [comment]
I have encountered the same problem and fixed it by changing the line
`qmi-firmware-update --reset -d "$deviceid"`
to
`qmi-firmware-update --reset -t /dev/ttyUSB2`

Please note, that the modem might have another device path for you. Check `dmesg` for more info.

For some reason, the other device selection options (`-s`, `-d`, `-w`) didn't work for me and failed with the mentioned error message:
`error: reseter operation finished: No devices found to run reset operation`

**Comment by jjhdtv** [comment]
Old thread but I had to em7455s that I used this method to restore to generic.  First one was successful.  Second the scripts hangs at the end after flashing SWI9X30C... and I get 

reseter operation finish successfully
waiting for modem in boothold mode....
waiting for modem in boothold mode

after hours of the same, need to ctrl c out it

I also notice a firmware mismatch in the "modem.log" file generated
fw version mismatch
carrier name mismatch
config name mismatch

Pounding my head against the wall, hope to get some help   thank you

**Reply by kneave** [reply]
I had this issue too, I hope you found a soloution but in case you didn't or for others who end up here here's what I did. I nothiced I wasn't getting any USB serial ports so thought I'd try forcing QMI mode.
`sudo qmicli --device-open-mbim -p -d /dev/cdc-wdm0 --dms-swi-set-usb-composition=6`

then to reset the modem

`sudo qmicli --device-open-mbim -p -d /dev/cdc-wdm0 --dms-set-operating-mode=offline`
`sudo qmicli --device-open-mbim -p -d /dev/cdc-wdm0 --dms-set-operating-mode=reset`

I then ran `autoflash-7455.sh` and it completed fine, after that I manually switched to MBIM mode as it's what I need:
`sudo qmicli --device-open-mbim -p -d /dev/cdc-wdm0 --dms-swi-set-usb-composition=8`

Seems to have worked

**Comment by Rudder2** [comment]
I read here https://github.com/danielewood/sierra-wireless-modems/discussions/71 that suspending his computer made it work.  I Hibernated my computer and woke it up and the script finished successfully!  Hope this helps someone else!
